src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test/OpenDocumentGeneratorTest.cs

```diff
@@ -142,8 +141,14 @@ await projectManager.UpdateAsync(updater =>
             updater.DocumentAdded(_hostProject1.Key, _documents[0], _documents[0].CreateEmptyTextLoader());
 
             // Act
-            updater.ProjectWorkspaceStateChanged(_hostProject1.Key,
-                ProjectWorkspaceState.Create(LanguageVersion.CSharp8));
+            var changed = _hostProject1 with
+            {
+                Configuration = _hostProject1.Configuration with
+                {
+                    CSharpLanguageVersion = CodeAnalysis.CSharp.LanguageVersion.CSharp8
+                }
+            };
```

Comment:
I'm pretty sure you can use dotted names in `with`. Doesn't block this PR, but good for a follow up.

```suggestion
            var changed = _hostProject1 with
            {
                Configuration.CSharpLanguageVersion = CodeAnalysis.CSharp.LanguageVersion.CSharp8
            };
```

Reply:
Actually, it looks like you can't do this, but I was sure we added it. Sigh...